src/extension/background-script/connectors/alby.ts

```diff
@@ -0,0 +1,378 @@
+      });
+      // TODO: make authorize URL in alby-js-sdk customizable
+      if (process.env.ALBY_OAUTH_AUTHORIZE_URL) {
+        authUrl = authUrl.replace(
+          "https://getalby.com/oauth",
+          process.env.ALBY_OAUTH_AUTHORIZE_URL
+        );
+      }
+      authUrl += "&webln=false"; // stop getalby.com login modal launching lnurl auth
+      const authResult = await this.launchWebAuthFlow(authUrl);
+      const code = new URL(authResult).searchParams.get("code");
+      if (!code) {
+        throw new Error("Authentication failed: missing authResult");
+      }
+
+      const token = await authClient.requestAccessToken(code);
+      await this._updateOAuthToken(token.token);
+      return authClient;
+    } catch (error) {
+      console.error(error);
+      throw error;
+    }
+  }
+
+  async launchWebAuthFlow(authUrl: string) {
+    const authResult = await browser.identity.launchWebAuthFlow({
+      interactive: true,
+      url: authUrl,
+    });
+
+    return authResult;
+  }
+
+  async getAccessToken(authResult: string, authClient: auth.OAuth2User) {
+    const authToken = new URL(authResult).searchParams.get("code");
+    if (!authToken) {
+      throw new Error("Authentication failed: missing token");
+    }
+
+    await authClient.requestAccessToken(authToken);
+  }
+
+  generateHmacVerification(uri: string) {
+    const mac = hmacSHA256(uri, HMAC_VERIFY_HEADER_KEY).toString(Base64);
+    return encodeURIComponent(mac);
+  }
+
+  async request<Type>(
+    method: Method,
+    path: string,
+    args?: Record<string, unknown>
+  ): Promise<Type> {
+    throw new Error(
+      "Unsupported. Migrate to alby-js-sdk and remove this function"
+    );
+  }
+
+  private async _getAlbyClient(): Promise<Client> {
+    if (this._clientPromise) {
+      return this._clientPromise;
+    }
+
+    this._clientPromise = new Promise<Client>((resolve, reject) => {
+      (async () => {
+        try {
+          this._authUser = await this.authorize();
+          resolve(new Client(this._authUser, this._getRequestOptions()));
+        } catch (error) {
+          reject(error);
+        }
+      })();
+    });
+    return this._clientPromise;
+  }
+
+  private async _request<T>(func: (client: Client) => T) {
+    const client = await this._getAlbyClient();
+    if (!this._authUser) {
+      throw new Error("this._authUser was not created");
+    }
+    const oldToken = this._authUser?.token;
+    let result: T;
+    try {
+      result = await func(client);
+    } catch (error) {
+      console.error(error);
+      throw error;
+    } finally {
+      const newToken = this._authUser.token;
+      if (newToken && newToken !== oldToken) {
+        await this._updateOAuthToken(newToken);
+      }
+    }
+    return result;
+  }
+
+  private _getRequestOptions(): Partial<RequestOptions> {
+    return {
+      ...(process.env.ALBY_API_URL
+        ? { base_url: process.env.ALBY_API_URL }
+        : {}),
+    };
```

Comment:
Should we pass a user_agent here? requires a new deployment of alby-js-sdk

Reply:
Yes, please add.

Reply:
@bumi could you let me know when you have deployed a new version of alby-js-sdk?